src/main/java/com/synopsys/integration/alert/channel/event/DistributionEvent.java

```diff
@@ -23,12 +23,17 @@
  */
 package com.synopsys.integration.alert.channel.event;
 
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 import com.synopsys.integration.alert.common.event.ContentEvent;
 import com.synopsys.integration.alert.common.model.AggregateMessageContent;
 
 public abstract class DistributionEvent extends ContentEvent {
     private final Long commonDistributionConfigId;
-    private Long auditEntryId;
+    private Map<Long, Long> notificationIdToAuditId;
```

Comment:
We have a map now because AggregateMessageContent can include multiple Notifications now, and we still want an audit for each of the notifications, otherwise when you resend a notification that was sent in a AggregateMessageContent with other notifications the 'Last Sent' time gets updated for all the notifications that were in that initial AggregateMessageContent